class.c

```diff
@@ -1269,6 +1269,60 @@ rb_mod_ancestors(VALUE mod)
     return ary;
 }
 
+static void
+collect_descendants(VALUE mod, VALUE hash)
+{
+    if (BUILTIN_TYPE(mod) != T_ICLASS) {
+        rb_hash_aset(hash, mod, Qnil);
+    }
+
+    rb_class_foreach_subclass(mod, collect_descendants, hash);
```

Comment:
Does `rb_class_foreach_subclass` work for modules too?

Reply:
Yes.
https://github.com/ruby/ruby/pull/3471/files#diff-02bfcb686555945e00ad1d80a1ece424R225-R233